ion/tokenizer.go

```diff
@@ -1379,3 +1383,15 @@ func isNewLineChar(c int) bool {
 	return c == 0x0A || //new line
 		c == 0x0D //carriage return
 }
+
+func isClobWithInvalidChar(clob bool, c int) bool {
+	// The string in clob may only contain 7-bit ASCII characters.
+	return clob && c > 0x7f
```

Comment:
[The clob section](https://amzn.github.io/ion-docs/docs/spec.html#clob) of the primary spec page says:

> The string may only contain legal 7-bit ASCII characters, using the same escaping syntax as string and symbol values.

However, there's an important clarification buried on the [Strings and Clobs](https://amzn.github.io/ion-docs/docs/stringclob.html#ion-clob) page:

> Non-printable ASCII and non-ASCII Unicode code points are not allowed un-escaped in the string bodies.

This means that ASCII characters below `31` (`0x1F`) must also be escaped because they are non-printable. The same page provides a listing of alternative escapes (`BEL` / `0x07` / `\a`, for example).

Reply:
We have already done that in an earlier [pull request](https://github.com/amzn/ion-go/pull/58/commits/460b748701e995b6754e8ad29726b98a375644c8); it is [```isProhibitedControlChar()```](https://github.com/amzn/ion-go/blob/master/ion/tokenizer.go#L1361) function.

Both of these checks are done on both [```readString()```](https://github.com/amzn/ion-go/pull/130/files#diff-99145fa564e42a4a98e734a9e12a7c55R586) and [```readLongString()```](https://github.com/amzn/ion-go/pull/130/files#diff-99145fa564e42a4a98e734a9e12a7c55R630).

I did not combine them into one function, as my understanding was ```string``` can hold characters beyond ```\x7f``` - Unless I am wrong. I do not see any mention of 7-bit ASCII limitation for string in the doc and also looked at some of the string [examples in good directory](https://github.com/amzn/ion-tests/blob/master/iontestdata/good/strings.ion#L9) where they contain 8-bit characters.

Reply:
> We have already done that in an earlier pull request; it is isProhibitedControlChar() function.

Thanks for pointing that out. PR reviews always involve looking at the codebase through a keyhole -- only the items in the diff are brought to my attention. 

I'd still change the name of `isClobWithInvalidChar`. It's really checking to see if the current character is ASCII or not, which is a subset of checking for clob character validity. Maybe `isClobWithNonAsciiChar`?

> I did not combine them into one function, as my understanding was string can hold characters beyond \x7f - Unless I am wrong.

No, you're right. The `string` type holds unicode code points and so can store non-ascii character data. I don't think you need to combine the two functions, I just think the `isClobWithInvalidChar` function should have a name which describes its purpose more crisply.

Reply:
Will update the name to ```isClobWithNonAsciiChar```

I think we can close #130?

Reply:
> I think we can close #130?

#130 is this PR. Did you mean to post a different number?

Reply:
I meant this issue: https://github.com/amzn/ion-docs/issues/130

Reply:
Oh, sorry. I just noticed the issue is opened in ion-doc repo

Reply:
Oh! It's the same issue number in a different repo. What a fun coincidence.

> Oh, sorry. I just noticed the issue is opened in ion-doc repo

Yeah, the `ion-docs` issue is reorganize that section of the spec so the "non-displayable" restriction on clob ASCII characters is more easily discoverable.